### ISSUE TITLE
How many epoches for each trail?

### ISSUE THREAD

**Issue by 0i0i0i** [microsoft/nni#5703]
**Describe the issue**: I followed the examples/tutorials/hpo_quickstart_pytorch and modify it to optimize for some regression model. In each trail, I set up an epoch loop to 1000 epoches and report the itermidiaate losses in every iteration. Does the tuner wait till the end of each trail and evaluate its metric or does it stop it earlier if I believes it had enough for that trail? Which document describe this? 

**Environment**:
- NNI version: 2.7
- Training service (local|remote|pai|aml|etc): local
- Client OS:ubuntu 2204
- Server OS (for remote mode only):
- Python version: 3.10
- PyTorch/TensorFlow version: pytorch 2.1
- Is conda/virtualenv/venv used?: conda
- Is running in Docker?:no 

**Configuration**:
 - Experiment config (remember to remove secrets!):
 - Search space:

**Log message**:
 - nnimanager.log:
 - dispatcher.log:
 - nnictl stdout and stderr:
 

**How to reproduce it?**:
